api/app/src/command/medical/document/document-query.ts

```diff
@@ -186,3 +151,58 @@ export const updateDocQuery = async ({
     transaction && (await transaction.commit());
   }
 };
+
+const setDocQueryProgress = ({
+  patient,
+  downloadProgress,
+  convertProgress,
+  convertResult,
+  restart,
+}: Omit<UpdateDocQueryParams, "patient"> & { patient: Patient }): DocumentQueryProgress => {
+  if (restart) {
+    return { download: downloadProgress };
+  }
+
+  const patientDocProgress = patient.data.documentQueryProgress;
+
+  const docQueryProgress = {
+    ...patientDocProgress,
+  };
+
+  if (downloadProgress) {
+    docQueryProgress.download = {
+      ...patientDocProgress?.download,
+      ...downloadProgress,
+    };
+  }
+
+  if (convertProgress) {
+    docQueryProgress.convert = {
+      ...patientDocProgress?.convert,
+      ...convertProgress,
+    };
+  }
+
+  if (convertResult) {
+    const successfulConvert = patientDocProgress?.convert?.successful ?? 0;
+    const errorsConvert = patientDocProgress?.convert?.errors ?? 0;
+    const totalToConvert = patientDocProgress?.convert?.total ?? 0;
+    const docQueryProgressStatus = successfulConvert + errorsConvert + 1 >= totalToConvert;
```

Comment:
Why the +1?

Reply:
Because `successfulConvert + errorsConvert` is what is currently in the db. So i need to add one to reflect the update being made.